### PR TITLE
Improve representation of variables rendered by the debugger

### PR DESCRIPTION
**Pull request by dlsniper**
This attempts to make variables to be displayed nicer when debugging is being used.
It also adds some clarity for numeric types as right now it's hard to distinguish between int and uint64 for example.

Before:
image

After:
image

### CODE REVIEW

**Comment by dlsniper** [comment]
At the moment complex types such as channels, maps, slices and structs are not supported yet by delve but once they are, I'll work on those as well.

**Comment by dlsniper** [comment]
@ignatov what do you think, could this be merged as well? Thank you.

**Comment by ignatov** [comment]
What about pointer?
screenshot 2015-09-20 23 17 22

File: src/com/goide/dlv/DlvStackFrame.java [hunk 1]
```diff
@@ -148,20 +147,65 @@ public void computePresentation(@NotNull XValueNode node, @NotNull XValuePlace p
...
       }
 
       @Nullable
       private XValuePresentation getPresentation() {
-        String type = variable.type;
+        final String type = variable.type;
         final String value = variable.value;
-        if (NUMBERS.contains(type)) return new XNumericValuePresentation(value);
-        if ("struct string".equals(type)) return new XStringValuePresentation(value);
+        if (NUMBERS.contains(type)) return new XValuePresentation() {
```
Comment (ignatov) [inline src/com/goide/dlv/DlvStackFrame.java]: Why do you expand this XNumericValuePresentation(value)?

Comment (dlsniper) [inline src/com/goide/dlv/DlvStackFrame.java]: For the type of the value. Right now one can't distinguish between int, uint, int64 and so on, this should help out, see the screenshot.

File: src/com/goide/dlv/DlvStackFrame.java [hunk 2]
```diff
@@ -148,20 +147,65 @@ public void computePresentation(@NotNull XValueNode node, @NotNull XValuePlace p
...
+            @NotNull
+            @Override
+            public String getType() {
+              return type;
+            }
+
             @Override
             public void renderValue(@NotNull XValueTextRenderer renderer) {
-              renderer.renderValue(value);
+              renderer.renderValue(value, GoSyntaxHighlightingColors.BUILTIN_TYPE_REFERENCE);
```
Comment (dlsniper) [inline src/com/goide/dlv/DlvStackFrame.java]: @ignatov I'm not sure about this change, I can revert it if needed.

**Comment by ignatov** [comment]
This PR provides a lot of unused information about types for primitive values. Even in Java we don't show such information yet.
BTW, I'm looking forward to https://groups.google.com/forum/#!topic/delve-dev/Z0lDomkS9LI.

File: src/com/goide/dlv/DlvStackFrame.java (hunk 1 above)
Comment (ignatov) [inline src/com/goide/dlv/DlvStackFrame.java]: Such type information adds lots of visual noise. Even in Java we don't have such visible information and we don't have any claim about it for many years.
screenshot 2015-09-22 11 17 53

**Pull request closed without merging**